web/src/buddy_data.ts

```diff
@@ -381,8 +384,9 @@ function filter_user_ids(user_filter_text: string, user_ids: number[]): number[]
             return false;
         }
 
-        if (!people.is_person_active(user_id)) {
-            // Deactivated users are hidden from the right sidebar entirely.
+        const is_dm = narrow_state.pm_ids_set().has(user_id);
+        if (!people.is_person_active(user_id) && !is_dm) {
+            // Deactivated users are hidden from the right sidebar except for the dm.
```

Comment:
I'd phrase this a little clearer as "Deactivated users are hidden in the buddy list except in DM narrows."